src/components/Wallet.tsx

```diff
@@ -150,7 +150,12 @@ const Wallet: React.FC<RouteComponentProps> = (props: WalletProps) => {
       toast.error('Error disconnecting wallet')
     } finally {
       setLoadingLabel(undefined)
-      props.history.push('/')
+
+      // If on trade page already, keep pre-selected values when logging out
+      const pathname = props.location.pathname
```

Comment:
nit: `const { pathname } = props.location`

Reply:
Since we are slowly switching to hooks for react-router. Mind changing Wallet from `withRouter` to hooks?

Reply:
> nit: const { pathname } = props.location

Done

> Since we are slowly switching to hooks for react-router. Mind changing Wallet from withRouter to hooks?

Should be in a different PR. Will create a task for it.